chunk_utils.py

```
@@ -0,0 +1,209 @@
+            sections.append((current_heading, current_section))
+        
+        # Now create chunks from sections
+        chunks = []
+        
+        # Parse domain for metadata
+        domain = urlparse(source_url).netloc
+        
+        for section_heading, section_content in sections:
+            # If the section is smaller than chunk_size, keep it as one chunk
+            if len(section_content) <= self.chunk_size:
+                chunk_id = hashlib.md5(f"{source_url}:{section_heading}".encode()).hexdigest()
```

Ruff (0.8.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)